ntbs-service/Services/AlertService.cs

```diff
@@ -129,19 +130,12 @@ public async Task DismissAllOpenAlertsForNotification(int notificationId)
 
         private async Task DismissDuplicationAlertsOnDuplicateRecord(int notificationId, int duplicateAlertId)
         {
-            var duplicateRecordAlerts = await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId);
-            if (!(duplicateRecordAlerts is null))
-            {
-                foreach (
-                    var alert in duplicateRecordAlerts
-                        .Where(al => al is DataQualityPotentialDuplicateAlert duplicateAlert && 
-                                     duplicateAlert.DuplicateId == notificationId)
-                         )
-                {
-                    // Only want to dismiss duplicate alerts where the duplicateId is equal to the original notificationId being closed
-                    SetClosedAndAddClosureDate(alert);
-                }
-            }
+            var duplicateRecordAlerts = (await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId))?
+                .Where(al =>
+                    al is DataQualityPotentialDuplicateAlert duplicateAlert &&
+                    duplicateAlert.DuplicateId == notificationId);
+            
+            duplicateRecordAlerts?.ForEach(a => SetClosedAndAddClosureDate(a));
```

Comment:
I generally don't like `.ForEach` because it gives the appearance of a functional style of programming, but it's really entirely object-oriented. (I don't know how much you know about this distinction, but in a functional style you don't change existing objects, you return new ones instead... foreach doesn't return anything, it exists only to change existing objects).

So I think it's better to go with a standard `foreach` loop in general.

(It's worth stating that other people do have different opinions about this).